lib/ansible/galaxy/collection/__init__.py

```diff
@@ -743,7 +743,8 @@ def install_collections(
 
     keyring_exists = artifacts_manager.keyring is not None
     with _display_progress("Starting collection install process"):
-        for fqcn, concrete_coll_pin in dependency_map.items():
+        for fqcn in dependency_map.keys():
+            concrete_coll_pin = dependency_map[fqcn]
```

Comment:
Why is this change needed?

Reply:
To avoid mutating the dictionary while iterating over it.

Reply:
I guess that's not necessary.